server/src/com/google/refine/Refine.java

```diff
@@ -96,7 +99,7 @@ public static void main(String[] args) throws Exception {
         // System.setProperty("debug","true");
 
         // set the log verbosity level
-        org.apache.log4j.Logger.getRootLogger().setLevel(Level.toLevel(Configurations.get("refine.verbosity", "info")));
+        setRootLoggerLevel(Configurations.get("refine.verbosity"));
```

Comment:
Hello @tfmorris I agree, Thanks for your suggestion. Was unaware of the setAllLevels method.

Although there is one catch; using the above suggested implementation would never allow the application to behave as per the configuration in log4j.properties making it necessary to supply an argument and forcibly use that level in all classes (even where it might not even be required).

Also if we do not supply any argument, the level defaults to DEBUG for the entire application, which as you rightly pointed would create a lot of noise.

What do you think about checking if the argument is supplied or not before doing the setAllLevels?

```
String logLevelArg = Configurations.get("refine.verbosity");
if (logLevelArg != null && !logLevelArg.isEmpty()) {
     Configurator.setAllLevels(LogManager.getRootLogger().getName(),Level.toLevel(Configurations.get("refine.verbosity", "info")));
}
```

If it is, we know the user wants to forcibly set level to what he wishes for the entire app.
And if it is not, the configured levels in log4j2.properties would be set for each class (including the rootLogger which is INFO).

Reply:
That sounds great

Reply:
Thankyou! I have made the change.